src/main/kotlin/domain/card/Cards.kt

```diff
@@ -0,0 +1,63 @@
+package domain.card
+
+class Cards(cards: List<Card> = emptyList()) {
+
+    private val cards: MutableList<Card> = cards.toMutableList()
+
+    fun add(card: Card) {
+        cards.add(card)
+    }
+
+    fun current(): Set<Card> {
+        return cards.toSet()
+    }
+
+    fun result(): Int {
+        val sumWithoutAce = sumWithoutAce()
+
+        if (doesNotHaveAce()) {
+            return sumWithoutAce
+        }
+
+        val sumOfAllCases = sumOfAceCases().map {
+            sumWithoutAce + it
+        }
+
+        if (sumOfAllCases.contains(BLACKJACK_POINT)) {
+            return BLACKJACK_POINT
+        }
+
+        return sumOfAllCases.min()
+    }
+
+    private fun doesNotHaveAce(): Boolean {
+        return numberOfAce() == 0
+    }
+
+    private fun sumWithoutAce(): Int {
+        return cards.filterNot {
+            it.denomination == Denomination.ACE
+        }.sumOf {
+            it.numbers.first()
+        }
+    }
+
+    private fun numberOfAce(): Int {
+        return cards.filter {
+            it.denomination == Denomination.ACE
+        }.size
+    }
```

Comment:
count를 활용해도 좋을 것 같아요.

```suggestion
    private fun numberOfAce(): Int {
        return cards.filter {
            it.denomination == Denomination.ACE
        }.size
    }
    private fun numberOfAce(): Int = cards.count {
        it.denomination == Denomination.ACE
    }
```